packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -245,16 +245,20 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
     };
 
     let onBlur = (e) => {
-      let isInAnyScope = isElementInAnyScope(e.relatedTarget, scopes);
-
-      if (!isInAnyScope) {
-        activeScope = scopeRef;
-        focusedNode.current = e.target;
-        // Firefox doesn't shift focus back to the Dialog properly without this
-        raf.current = requestAnimationFrame(() => {
-          focusedNode.current.focus();
-        });
-      }
+      // Brief timeout for document.activeElement to update (e.g. user clicks inside a iframe in a dialog)
```

Comment:
timeout? or should this be a raf?
I imagine we should have something that can cancel the raf/timeout so we can avoid race conditions like the raf.current already has

Reply:
if it was an issue with raf not being advance-able, take a look at https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/dialog/test/DialogTrigger.test.js#L40